File: app/src/androidTest/java/com/pavelnazimok/uitesting/java/robots/MenuRobot.java
package com.pavelnazimok.uitesting.java.robots;

import com.pavelnazimok.uitesting.R;

import static com.schibsted.spain.barista.interaction.BaristaClickInteractions.clickOn;

@SuppressWarnings("UnusedReturnValue")
public class MenuRobot {

    public MenuRobot clickFirstScreen() {
        clickOn(R.id.nav_first);
        return this;
    }

    public MenuRobot clickSecondScreen() {
        clickOn(R.id.nav_second);
        return this;
    }

    public MenuRobot clickThirdScreen() {
        clickOn(R.id.nav_third);
        return this;
    }
}

File: app/src/androidTest/java/com/pavelnazimok/uitesting/kotlin/robots/MenuRobot.kt
package com.pavelnazimok.uitesting.kotlin.robots

import com.pavelnazimok.uitesting.R
import com.pavelnazimok.uitesting.kotlin.utils.waitForView
import com.schibsted.spain.barista.interaction.BaristaClickInteractions.clickOn

fun onMenu(func: MenuRobot.() -> Unit): MenuRobot {
    return MenuRobot().apply {
        assertMenuIsDisplayed()
        func()
    }
}

class MenuRobot {

    fun clickFirstScreen() {
        clickOn(R.id.nav_first)
    }

    fun clickSecondScreen() {
        clickOn(R.id.nav_second)
    }

    fun clickThirdScreen() {
        clickOn(R.id.nav_third)
    }

    fun assertMenuIsDisplayed() {
        waitForView(R.id.nav_bottom_bar)
    }
}
